File: PST/FUEL Source Code/FUEL/Backup/CheckForUpdates.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.CheckForUpdates
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using FUEL.FS;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using RDL;
using System;
using System.Diagnostics;
using System.Reflection;

namespace FUEL
{
  public class CheckForUpdates
  {
    private string _DLLoc { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private string _SaveAs { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private CheckForUpdates.CheckType _FUELType { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private Version _ServerVersion { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private Version _CurrentVersion { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private Version _RequiredPSTVersion { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private Version _RequiredFSVersion { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private CheckForUpdates.UpdateType _UpdateRequired { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private string _FileList { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public object UpdateAvailable => (object) this._UpdateRequired;

    public string ServerVersion => this._ServerVersion.ToString();

    public string CurrentVersion => this._CurrentVersion.ToString();

    public CheckForUpdates(string DLLocation, string SaveAs, CheckForUpdates.CheckType FUELType)
    {
      this._DLLoc = DLLocation;
      this._SaveAs = SaveAs;
      this._FUELType = FUELType;
      if (this.DownloadVersionFile())
      {
        this.ReadVersionFile();
        this.DetermineIfUpdateNeeded();
      }
      else
        this._UpdateRequired = CheckForUpdates.UpdateType.None;
    }

    private bool DownloadVersionFile()
    {
      HTTPDownloadSync httpDownloadSync = new HTTPDownloadSync(this._DLLoc, this._SaveAs);
      httpDownloadSync.StartDownload();
      return Conversions.ToBoolean(httpDownloadSync.Success);
    }

    private void DetermineIfUpdateNeeded()
    {
      this._CurrentVersion = new Version(Assembly.GetExecutingAssembly().GetName().Version.ToString());
      if (this._CurrentVersion < this._RequiredPSTVersion & this._FUELType == CheckForUpdates.CheckType.PST)
        this._UpdateRequired = CheckForUpdates.UpdateType.RequiredUpdate;
      else if (this._CurrentVersion < this._RequiredFSVersion & this._FUELType == CheckForUpdates.CheckType.FS)
        this._UpdateRequired = CheckForUpdates.UpdateType.RequiredUpdate;
      else if (this._CurrentVersion < this._ServerVersion)
        this._UpdateRequired = CheckForUpdates.UpdateType.OptionalUpdate;
      else
        this._UpdateRequired = CheckForUpdates.UpdateType.None;
    }

    private void ReadVersionFile()
    {
      string[,] strArray = FileProcessing.ReadDelimitedFile(this._SaveAs, ":");
      int num = Information.UBound((Array) strArray);
      int index = 0;
      while (index <= num)
      {
        string upper = strArray[index, 0].ToUpper();
        if (Operators.CompareString(upper, "CurrentVersion".ToUpper(), false) == 0)
          this._ServerVersion = new Version(strArray[index, 1]);
        else if (Operators.CompareString(upper, "RequiredPSTVersion".ToUpper(), false) == 0)
          this._RequiredPSTVersion = new Version(strArray[index, 1]);
        else if (Operators.CompareString(upper, "RequiredFSVersion".ToUpper(), false) == 0)
          this._RequiredFSVersion = new Version(strArray[index, 1]);
        else if (Operators.CompareString(upper, "FileList".ToUpper(), false) == 0)
          this._FileList = strArray[index, 1];
        checked { ++index; }
      }
    }

    public enum UpdateType
    {
      None,
      OptionalUpdate,
      RequiredUpdate,
    }

    public enum CheckType
    {
      PST,
      FS,
    }
  }
}

File: PST/FUEL/src/FUEL/CheckForUpdates.vb
Imports FUEL.FS
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports RDL
Imports System
Imports System.Diagnostics
Imports System.Reflection
Imports System.Runtime.CompilerServices

Namespace FUEL
    Public Class CheckForUpdates
        ' Methods
        Public Sub New(ByVal DLLocation As String, ByVal SaveAs As String, ByVal FUELType As CheckType)
            Me._DLLoc = DLLocation
            Me._SaveAs = SaveAs
            Me._FUELType = FUELType
            If Not Me.DownloadVersionFile Then
                Me._UpdateRequired = UpdateType.None
            Else
                Me.ReadVersionFile
                Me.DetermineIfUpdateNeeded
            End If
        End Sub

        Private Sub DetermineIfUpdateNeeded()
            Me._CurrentVersion = New Version(Assembly.GetExecutingAssembly.GetName.Version.ToString)
            Me._UpdateRequired = If(Not ((Me._CurrentVersion < Me._RequiredPSTVersion) And (Me._FUELType = CheckType.PST)), If(Not ((Me._CurrentVersion < Me._RequiredFSVersion) And (Me._FUELType = CheckType.FS)), If((Me._CurrentVersion >= Me._ServerVersion), UpdateType.None, UpdateType.OptionalUpdate), UpdateType.RequiredUpdate), UpdateType.RequiredUpdate)
        End Sub

        Private Function DownloadVersionFile() As Boolean
            Dim sync As New HTTPDownloadSync(Me._DLLoc, Me._SaveAs)
            sync.StartDownload
            Return Conversions.ToBoolean(sync.Success)
        End Function

        Private Sub ReadVersionFile()
            Dim array(,) As String(0 To .,0 To .) = FileProcessing.ReadDelimitedFile(Me._SaveAs, ":")
            Dim num2 As Integer = Information.UBound(array, 1)
            Dim num As Integer = 0
            Do While True
                Dim num3 As Integer = num2
                If (num > num3) Then
                    Return
                End If
                Dim str As String = array(num, 0).ToUpper
                If (str = "CurrentVersion".ToUpper) Then
                    Me._ServerVersion = New Version(array(num, 1))
                ElseIf (str = "RequiredPSTVersion".ToUpper) Then
                    Me._RequiredPSTVersion = New Version(array(num, 1))
                ElseIf (str = "RequiredFSVersion".ToUpper) Then
                    Me._RequiredFSVersion = New Version(array(num, 1))
                ElseIf (str = "FileList".ToUpper) Then
                    Me._FileList = array(num, 1)
                End If
                num += 1
            Loop
        End Sub


        ' Properties
        Private Property _DLLoc As String
            <DebuggerNonUserCode> _
            Get
                Return Me.__DLLoc
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String)
                Me.__DLLoc = AutoPropertyValue
            End Set
        End Property

        Private Property _SaveAs As String
            <DebuggerNonUserCode> _
            Get
                Return Me.__SaveAs
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String)
                Me.__SaveAs = AutoPropertyValue
            End Set
        End Property

        Private Property _FUELType As CheckType
            <DebuggerNonUserCode> _
            Get
                Return Me.__FUELType
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As CheckType)
                Me.__FUELType = AutoPropertyValue
            End Set
        End Property

        Private Property _ServerVersion As Version
            <DebuggerNonUserCode> _
            Get
                Return Me.__ServerVersion
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Version)
                Me.__ServerVersion = AutoPropertyValue
            End Set
        End Property

        Private Property _CurrentVersion As Version
            <DebuggerNonUserCode> _
            Get
                Return Me.__CurrentVersion
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Version)
                Me.__CurrentVersion = AutoPropertyValue
            End Set
        End Property

        Private Property _RequiredPSTVersion As Version
            <DebuggerNonUserCode> _
            Get
                Return Me.__RequiredPSTVersion
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Version)
                Me.__RequiredPSTVersion = AutoPropertyValue
            End Set
        End Property

        Private Property _RequiredFSVersion As Version
            <DebuggerNonUserCode> _
            Get
                Return Me.__RequiredFSVersion
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Version)
                Me.__RequiredFSVersion = AutoPropertyValue
            End Set
        End Property

        Private Property _UpdateRequired As UpdateType
            <DebuggerNonUserCode> _
            Get
                Return Me.__UpdateRequired
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As UpdateType)
                Me.__UpdateRequired = AutoPropertyValue
            End Set
        End Property

        Private Property _FileList As String
            <DebuggerNonUserCode> _
            Get
                Return Me.__FileList
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String)
                Me.__FileList = AutoPropertyValue
            End Set
        End Property

        Public ReadOnly Property UpdateAvailable As Object
            Get
                Return Me._UpdateRequired
            End Get
        End Property

        Public ReadOnly Property ServerVersion As String
            Get
                Return Me._ServerVersion.ToString
            End Get
        End Property

        Public ReadOnly Property CurrentVersion As String
            Get
                Return Me._CurrentVersion.ToString
            End Get
        End Property


        ' Fields
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __DLLoc As String
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __SaveAs As String
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __FUELType As CheckType
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __ServerVersion As Version
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __CurrentVersion As Version
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __RequiredPSTVersion As Version
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __RequiredFSVersion As Version
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __UpdateRequired As UpdateType
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __FileList As String

        ' Nested Types
        Public Enum CheckType
            ' Fields
            PST = 0
            FS = 1
        End Enum

        Public Enum UpdateType
            ' Fields
            None = 0
            OptionalUpdate = 1
            RequiredUpdate = 2
        End Enum
    End Class
End Namespace

